engine/grid.py

```diff
@@ -118,11 +134,17 @@ def generate_nodes(start_lat, start_long, rows, cols, step_size_m):
         self.active_waypoints_list = []
         self.inactive_waypoints_list = []
 
+        # Properties used to determine our adjustable grid traversal pathing
+        self.waypoints_is_finished = False
+        self.curr_pos = None
```

Comment:
might want to specify curr_pos is (row, col) if vertical and (col, row) if not vertical

Reply:
wdym by vertical? if direction is set to vertical?

Reply:
yeah, based on is_vertical

Reply:
added